neuraxle/metaopt/auto_ml.py

```diff
@@ -797,7 +817,21 @@ def split_data_container(data_container: DataContainer) -> Tuple[DataContainer,
         validation_data_container = DataContainer(data_inputs=validation_data_inputs,
                                                   expected_outputs=validation_expected_outputs)
 
-        return train_data_container, validation_data_container
+        for (train_current_id, train_di, train_eo), (validation_current_id, validation_di, validation_eo) in \
+                zip(train_data_container, validation_data_container):
```

Comment:
a comment to explain what this loop does and how it changes the data would be nice.

Reply:
It creates a generator so that I can loop on the training, and validation data containers....

```
        validation_splits = self.validation_split_function(data_container)

        for trial_number in range(self.n_trial):
            auto_ml_data = AutoMLContainer(
                trial_number=trial_number,
                trials=self.hyperparams_repository.load_all_trials(TRIAL_STATUS.SUCCESS),
                hyperparameter_space=self.pipeline.get_hyperparams_space(),
                main_scoring_metric_name=self.trainer.get_main_metric_name()
            )

            with self.hyperparams_repository.new_trial(auto_ml_data) as repo_trial:
                for training_data_container, validation_data_container in validation_splits:

```

Reply:
add :return comment, and a comment above the loop